pkg/ccl/importccl/import_stmt.go

```diff
@@ -99,6 +99,8 @@ const (
 	avroSchema    = "schema"
 	avroSchemaURI = "schema_uri"
 
+	pgDumpIgnoreAllUnsupported = "ignore_unsupported"
```

Comment:
I think as a user I would find it confusing that there are statements which still return some sort of error like "unimplemented" with this flag (please correct me if I'm reading this wrong, but a statement `CREATE OPERATOR` is not "unsupported"). As you mentioned in the commit message, these would be documented but I still think it might be confusing?

Does it make sense to ignore all unimplemented nodes if we can write them to a rejected file?

Reply:
I'm a little confused, I'm going to try and categorize the types of errors we can see, and then we can decide what should be guarded by this flag.

1. There are statements like `CREATE OPERATOR` which return an unimplemented error on parsing. There are several of these in sql.y. This PR replaced some (previously caught by our regex) of those unimplemented errors to instead be backed by an unsupported node, so that when we see such a node on parse.Parse() we can skip over it.

I guess my first question is: do we want to skip **all** sql statements which are unimplemented in our parser, or do we want to have control and only skip a subset of those. This would mean for the rest, the user would see an unable to parse error.

If the answer to the above question is yes, I would need to wholesale change all the sql.y grammar rules to be backed by our unsupported node. Albeit this could be done by just changing the `unimplemented()` methods they currently call.

2. The second group of statements is the ones that have associated tree.Nodes. A statement with a tree.Node is either "supported" by IMPORT, "unsupported and ignored", or just "unsupported" and in the last case, we will throw an error. Here I think I agree with you that if we see something which made it past parsing, but we don't support it then it should not throw an error. This PR doesn't do that yet.

Let me know what you think or if I'm misunderstanding your comment. Going to cc: @mwang1026 in case he has thoughts too.

Reply:
I agree it's confusing UX for a user that said "I want to ignore things that are unsupported" for it to ignore just a subset, especially if we have a "reject" file

Reply:
> I guess my first question is: do we want to skip all sql statements which are unimplemented in our parser, or do we want to have control and only skip a subset of those.

I think that that's also what I was after. I think that moving the ignoring logic from regexes to sql.y is good and can be done on it's own. My concern was that before we were silently ignoring these, so it wasn't surprising to a user that they would run an import and reach some unsupported statement. However, now if they want these statements to be unsupported they have to run the import with the `ignore_unsupported`, and they might _still_ get "unsupported" errors which is confusing.

I don't think I have a good enough grasp of why some nodes are "unsupported and ignored" or just "unsupported" to have a more concrete opinion on the best way to resolve this though.

Reply:
@pbardea just to re-outline the new approach since there has been a lot of discussion:

The `ignore_unsupported` option will skip the following genres of statements:
- All statements that have rules in sql.y and return an `unimplemented` error.
- All statements that were previously ignored silently egs: `BEGIN`, `COMMIT` etc
- All statements that were previously successfully parsed, but returned an `unsupported` error during tree.Node evaluation.

The first commit only addresses the first bullet point. Subsequent commits will tackle the next two bullet points.